src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1465,6 +1465,31 @@
       (format "ALTER TABLE reports ADD COLUMN logs %s DEFAULT NULL" jsonb-type)
       (format "ALTER TABLE reports ADD COLUMN resources %s DEFAULT NULL" jsonb-type))))
 
+(defn add-expression-indexes-for-bytea-queries
+  []
+  (if (sutils/postgres?)
+    (jdbc/do-commands
+      "CREATE UNIQUE INDEX reports_hash_expr_idx ON reports(trim(leading '\\x' from hash::text))"
+      "ALTER TABLE REPORTS DROP CONSTRAINT reports_hash_key"
+      "DROP INDEX reports_transaction_uuid_idx"
+      "CREATE INDEX reports_uuid_expr_idx ON reports(CAST(transaction_uuid AS text))"
```

Comment:
I think we'll probably want to make tests for these in the rollup using @senior's migration diff'ing, to make sure they don't just disappear!